### DISCUSSION TITLE
How to reduce libmxnet.so file size

### DISCUSSION THREAD

**Discussion by bykim555** [apache/incubator-mxnet#19724] [Q&A] [answered]
OS: Ubuntu 18.04 64bit LTS
Anaconda3-2019.03-Linux-x86_64
conda env used, Python: 3.6.8

git clone --recursive https://github.com/apache/incubator-mxnet.git
Tried to build version mxnet1.8.0.rc2
ci/build.py -p android_armv7. And build success.

But libmxnet.so(1.03GB) size is very big.

How to reduce libmxnet.so file size.

**Comment by leezu** [comment, marked as answer]
`ci/build.py -p android_armv7` is not optimized for size. You should be able to reduce the size by adding `-DCMAKE_BUILD_TYPE=Release` option to the `cmake` call in `ci/docker/runtime_functions.sh`. You can also manually run `strip --strip-unneeded libmxnet.so` after the build.

Finally, if you only need to support a known set of networks / operations, you can delete unneeded operators in `src/operator` before the build.
